vault/logical_cubbyhole.go

```diff
@@ -140,17 +141,22 @@ func (b *CubbyholeBackend) handleList(
 	if req.ClientToken == "" {
 		return nil, fmt.Errorf("[ERR] cubbyhole list: Client token empty")
 	}
+
 	// List the keys at the prefix given by the request
 	keys, err := req.Storage.List(req.ClientToken + "/" + req.Path)
 	if err != nil {
 		return nil, err
 	}
 
-	strippedKeys := []string{}
-	for _, key := range keys {
-		strippedKeys = append(strippedKeys, strings.TrimPrefix(key, req.ClientToken+"/"))
+	// Strip the token; also, add the path for the same reason as in
+	// passthrough
+	strippedKeys := make([]string, len(keys))
+	for i, key := range keys {
+		strippedKeys[i] = req.MountPoint + req.Path + strings.TrimPrefix(key, req.ClientToken+"/")
 	}
 
+	sort.Strings(strippedKeys)
```

Comment:
If we expect keys to be sorted as part of the spec, we may as well handle this once in `logical. ListResponse` to avoid doing it everywhere.